src/main/scala/com/ing/wbaa/rokku/proxy/api/directive/ProxyDirectives.scala

```diff
@@ -32,55 +35,76 @@ object ProxyDirectives extends LazyLogging {
     if (httpHeader.is(AUTHORIZATION_HTTP_HEADER_NAME)) {
       val signerType = httpHeader.value().split(" ").headOption
       logger.debug("Signertype used: {}", signerType)
+      getAccessKeyFromAuthString(httpHeader.value(), signerType)
+    } else None
 
-      signerType match {
-        case Some("AWS4-HMAC-SHA256") =>
-          val credential =
-            """\S+ Credential=(\S+), """.r
-              .findFirstMatchIn(httpHeader.value())
-              .map(_ group 1)
+  private[this] def getAccessKeyFromAuthString(authString: String, signerType: Option[String]): Option[AwsAccessKey] = {
+    signerType match {
+      case Some("AWS4-HMAC-SHA256") =>
+        val credential =
+          """\S+ Credential=(\S+), """.r
+            .findFirstMatchIn(authString)
+            .map(_ group 1)
 
-          credential.flatMap(_.split("/").headOption).map(AwsAccessKey)
+        credential.flatMap(_.split("/").headOption).map(AwsAccessKey)
 
-        case Some("AWS") =>
-          val accessKey =
-            """AWS (\S+):\S+""".r
-              .findFirstMatchIn(httpHeader.value())
-              .map(_ group 1)
+      case Some("AWS") =>
+        val accessKey =
+          """AWS (\S+):\S+""".r
+            .findFirstMatchIn(authString)
+            .map(_ group 1)
 
-          accessKey.map(AwsAccessKey)
+        accessKey.map(AwsAccessKey)
 
-        case _ =>
-          logger.warn("The necessary information couldn't be extracted from the authorization header, " +
-            s"this could be caused by a signer type that we don't support yet...: {}", httpHeader)
-          None
+      case _ =>
+        logger.warn("The necessary information couldn't be extracted from the authorization header, " +
+          s"this could be caused by a signer type that we don't support yet...: {}", authString)
+        None
+    }
+  }
+
+  private[this] def extractAuthorizationS3(signerType: String, authString: String): Option[AwsAccessKey] = {
+    //to use the method getAccessKeyFromAuthString we need to add something at the beginning and colon after - maybe we need think about sth else (to have one method the extract auth)
+    val authStringWithPrefixAnsPostfix = "x Credential=" + authString + ", x"
+    getAccessKeyFromAuthString(authStringWithPrefixAnsPostfix, Some(signerType))
+  }
+
+  private[this] def extractAuthorizationS3: Directive1[AwsAccessKey] = {
```

Comment:
Could we make names a bit more descriptive, like `extractAuthorizationS3FromParameters` and `extractAuthorizationS3FromHeaders`